src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -493,11 +498,9 @@ private boolean perform(Run src, Run<?,?> dst, String expandedFilter, String exp
     public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
 
         public FormValidation doCheckProjectName(
-                @AncestorInPath AbstractProject anc, @QueryParameter String value) {
-            // Require CONFIGURE permission on this project
-            if (!anc.hasPermission(Item.CONFIGURE)) return FormValidation.ok();
```

Comment:
This would be a security regression. You need to retain `@AncestorInPath Job job` so you can still do a permission check on it.